plugins/search-backend/src/service/router.ts

```diff
@@ -154,9 +160,13 @@ export async function createRouter(
         const resultSet = await engine?.query(query, { token });
 
         res.send(filterResultSet(toSearchResults(resultSet)));
-      } catch (err) {
+      } catch (error) {
+        if (error instanceof MissingIndexError) {
```

Comment:
Recently learned that `instanceof` can be unreliable, especially when packages are duplicated across multiple versions.

Consider checking the error name as a string instead?